### PR TITLE
Move Android benchmarks from Pixel 6 to Pixel 8

### PR DESCRIPTION
**Pull request by pzread**
This change moves our Android CPU and GPU benchmarks from Pixel 6 to Pixel 8.

CPU benchmarks are also reconfigured as below:

- Single-thread benchmarks will run on the largest core on Pixel 8 (core 8)
- Multi-thread benchmarks will run with 5 threads on largest core + big cores (core 8 + core 4-7)

Since this is a major change on hardware and benchmark config, we will use new benchmark series to track them

benchmark-extra: android-cpu-dt-only

#15452

### CODE REVIEW

**Comment by pzread** [comment]
Hi @mariecwhite For multi-thread benchmarks I think we can try to use 5 cores, while your original proposal is with 4 medium cores. Let me know if there is a special reason we want to stick with 4 medium cores.

File: build_tools/python/e2e_test_framework/device_specs/pixel_8_pro_specs.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+DEVICE_NAME = "pixel-8-pro"
+
+BIG_CORES = common_definitions.DeviceSpec.build(
+    id=unique_ids.DEVICE_SPEC_MOBILE_PIXEL_8_PRO + "-big-core",
+    device_name=DEVICE_NAME,
+    architecture=common_definitions.DeviceArchitecture.ARMV8_2_A_GENERIC,
+    host_environment=common_definitions.HostEnvironment.ANDROID_ARMV8_2_A,
+    device_parameters=common_definitions.DeviceParameters(
+        cpu_params=common_definitions.CPUParameters(pinned_cores=[4, 5, 6, 7, 8])
```
Comment (mariecwhite) [inline build_tools/python/e2e_test_framework/device_specs/pixel_8_pro_specs.py:19]: Does this use taskset to pin the threads or `--task_topology_cpu_ids`?

Comment (pzread) [reply build_tools/python/e2e_test_framework/device_specs/pixel_8_pro_specs.py:19]: taskset currently, but it's the next step to move to `--task_topology_cpu_ids`.

**Comment by antiagainst** [comment]
This only touches CPU benchmarks? It would be nice to make the PR title clearer. Do you have plans for GPU benchmarks?

**Comment by pzread** [comment]
> This only touches CPU benchmarks? It would be nice to make the PR title clearer. Do you have plans for GPU benchmarks?

Oh I missed it. This is for both CPU and GPU benchmarks. We majorly want to migrate the CPU benchmarks but trying to not maintain more phones, so GPU benchmarks are migrated together.

**Comment by antiagainst** [comment]
@pzread: I don't see GPU raw results for pixel 8 [here](https://gist.github.com/iree-github-actions-bot/dd69009eb417cf9b8e7eccaf75bdd74c#raw-latencies). Is it to happen in a later revision of this pull request?

**Comment by pzread** [comment]
> @pzread: I don't see GPU raw results for pixel 8 [here](https://gist.github.com/iree-github-actions-bot/dd69009eb417cf9b8e7eccaf75bdd74c#raw-latencies). Is it to happen in a later revision of this pull request?

I just re-triggered the benchmarks and will wait for the results

**Review by antiagainst — APPROVED** [review]
Nice, thanks!

**Review by mariecwhite — APPROVED** [review]

**Comment by pzread** [comment]
Also @dcaballe FYI

**Comment by pzread** [comment]
It looks like there is a regression on GPT2 w/ ukernel. I'll check if it's because of some CPU feature flags

**Comment by pzread** [comment]
> It looks like there is a regression on GPT2 w/ ukernel. I'll check if it's because of some CPU feature flags

Filed a bug to track this issue: #16084

**Pull request merged**